File: www/test/cgi.php
<?php
// Set the HTTP response headers
header("HTTP/1.1 200 OK");
header("Content-type: text/html");

// Print the received data from POST request
$data = "";
if ($_SERVER['REQUEST_METHOD'] == "POST") {
    $data = file_get_contents('php://input');
}

// Get and print the value of the "HTTP_COOKIE" environment variable
$cookie = $_SERVER['HTTP_COOKIE'];

// Generate the HTML response
$html_response = "
<!DOCTYPE html>
<html>
<head>
    <title>Form Data</title>
    <style>
        body {
            text-align: center;
            font-family: Arial, Helvetica, sans-serif;
            background-color: #1d3557;
            color: #f1faee;
        }
    </style>
</head>
<body>
    <h1>CGI working good from PHP</h1>
    <p>Form: $data</p>
    <p>Cookie: $cookie</p>
</body>
</html>
";

// Output the HTML response
echo $html_response;
?>

File: www/test/cgi.py
#!/usr/bin/python

import sys
import os

print("HTTP/1.1 200 OK")
print("Content-type: text/html\r\n")

# Print the received data to stdout
data = ""
if os.environ.get('REQUEST_METHOD') == "POST":
	data = sys.stdin.read()
Cookie = os.environ.get("HTTP_COOKIE")

html_response = f"""
<!DOCTYPE html>
<html>
<head>
    <title>Form Data</title>
	<style>
		body {{
			text-align: center;
			font-family: Arial, Helvetica, sans-serif;
			background-color: #1d3557;
			color: #f1faee;
		}}
	</style>
</head>
<body>
    <h1>CGI working good from Python</h1>
    <p>Form: {data}</p>
    <p>Cookie: {Cookie}</p>
</body>
</html>
"""
print(html_response)
